### PR TITLE
Certs

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
Use of trusted peer certificates.

This allows the feature of loading in a trusted peer certificate for use when verifying a peer that is connecting. Build option for enabling this feature is WOLFSSL_TRUST_PEER_CERT. An example of its use in both example/server/server.c and example/client/client.c. For both example server and example client a new flag '-E' has been added to allow for specifying the path to a trusted peer cert for loading. When enabled automated testing is performed by running scripts/trusted_peer.test.

### CODE REVIEW

File: examples/client/client.c [hunk 1]
```diff
@@ -380,6 +380,9 @@ static void Usage(void)
 #ifdef HAVE_CRL
     printf("-C          Disable CRL\n");
 #endif
+#ifdef WOLFSSL_TRUST_PEER_CERT
+    printf("-T <file>   Path to load trusted peer cert\n");
```
Comment (dgarske) [inline examples/client/client.c]: Looks like this should be "-E"?

**Comment by JacobBarthelmeh** [comment]
Thanks David, Yes it should print out -E. I wanted -T originally but it was already in use, this is a slip up from changing over to be -E.

**Comment by toddouska** [comment]
Please make changes discussed during review.

**Comment by toddouska** [comment]
Thanks for the changes, looks good.

**Pull request merged**